ov/logical_interconnects.go

```diff
@@ -0,0 +1,834 @@
+	if err := json.Unmarshal([]byte(data), &ethernetSettings); err != nil {
+		return ethernetSettings, err
+	}
+	return ethernetSettings, nil
+}
+
+func (c *OVClient) GetLogicalInterconnectFirmware(Id string) (Firmware, error) {
+	var (
+		uri      = "/rest/logical-interconnects/"
+		firmware Firmware
+	)
+	uri = uri + Id + "/firmware"
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return firmware, err
+	}
+	log.Debugf("GetLogicalInterconnect Firmware %s", data)
+	if err := json.Unmarshal([]byte(data), &firmware); err != nil {
+		return firmware, err
+	}
+	return firmware, nil
+}
+
+func (c *OVClient) GetLogicalInterconnectSNMPConfiguration(Id string) (SnmpConfiguration, error) {
+	var (
+		uri               = "/rest/logical-interconnects/"
+		snmpConfiguration SnmpConfiguration
+	)
+	uri = uri + Id + "/snmp-configuration"
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return snmpConfiguration, err
+	}
+	log.Debugf("GetLogicalInterconnect SNMP Configuration %s", data)
+	if err := json.Unmarshal([]byte(data), &snmpConfiguration); err != nil {
+		return snmpConfiguration, err
+	}
+	return snmpConfiguration, nil
+}
+func (c *OVClient) GetLogicalInterconnectForwardingInformationByMacAddress(MacAddress string, Id string) (InterconnectFibDataEntry, error) {
+	var (
+		InterconnectFibDataEntry InterconnectFibDataEntry
+	)
+	InterconnectFibData, err := c.GetLogicalInterconnectForwardingInformation(fmt.Sprintf("macAddress='%s'", MacAddress), "", Id)
+	if InterconnectFibData.Count > 0 {
+		return InterconnectFibData.Members[0], err
+	} else {
+		return InterconnectFibDataEntry, err
+	}
+
+}
+func (c *OVClient) GetLogicalInterconnectForwardingInformationByInternalVlan(InternalVlan string, Id string) (InterconnectFibDataEntry, error) {
+	var (
+		InterconnectFibDataEntry InterconnectFibDataEntry
+	)
+	InterconnectFibData, err := c.GetLogicalInterconnectForwardingInformation(fmt.Sprintf("internalVlan='%s'", InternalVlan), "", Id)
+	if InterconnectFibData.Count > 0 {
+		return InterconnectFibData.Members[0], err
+	} else {
+		return InterconnectFibDataEntry, err
+	}
+}
+func (c *OVClient) GetLogicalInterconnectForwardingInformationByInterconnectAndExternalVlan(InterconnectURI string, ExternalVlan string, Id string) (InterconnectFibDataEntry, error) {
+	var (
+		InterconnectFibDataEntry InterconnectFibDataEntry
+	)
+	InterconnectFibData, err := c.GetLogicalInterconnectForwardingInformation(fmt.Sprintf("interconnectUri='%s'", InterconnectURI), fmt.Sprintf("externalVlan='%s'", ExternalVlan), Id)
+	if InterconnectFibData.Count > 0 {
+		return InterconnectFibData.Members[0], err
+	} else {
+		return InterconnectFibDataEntry, err
+	}
+}
+
+func (c *OVClient) GetLogicalInterconnectForwardingInformation(filter1 string, filter2 string, Id string) (InterconnectFibData, error) {
```

Comment:
the api says that filter is an array, so you sgould use an array instead of limiting it upto 2 filters